media/js/base/mozilla-fxa-link.js

```diff
@@ -34,21 +41,68 @@ if (typeof window.Mozilla === 'undefined') {
         return href;
     };
 
+    /**
+     * Returns the hostname for a given URL.
+     * @param {String} url.
+     * @returns {String} hostname.
+     */
+    FxaLink.getHostName = function(url) {
+        var matches = url.match(/^https?:\/\/(?:[^/?#]+)(?:[/?#]|$)/i);
+        return matches && matches[0];
+    };
+
+    /**
+     * Intercept event handler for FxA links and forms, lets the browser drive the FxA Flow using
+     * the `showFirefoxAccounts` UITour API. Attaches several UTM parameters from the current page
+     * that will be forwarded to the browser and later on to FxA services.
+     * @param event {Event}
+     * @private
+     */
+    FxaLink.interceptFxANavigation = function(event) {
+        event.preventDefault();
+        var search = new URL(event.target.href).search;
+        var urlParams = new window._SearchParams(search).params;
+        var extraURLParams = {};
+        var allowedChars = /^[\w/.%-]+$/;
+        var knownParams = ['flow_id', 'flow_begin_time', 'device_id',
```

Comment:
Hmm - but when I change the above locally, the call to `showFirefoxAccounts()` fails silently. I wonder, is this because the required format has changed?

If that's the case would we need to change this convention in our helpers site-wide? How would that effect older clients?

Reply:
Yeah to match the `utm_*` parameters, this followed with snake_case. hmmm

Reply:
I have adjusted the function to convert the values for the proper case. Was there anything more you had in mind for utm_ value validation? The `getAttributionData` function does a bit of validation already: 

https://github.com/mozilla/bedrock/blob/9e6ed7eaa873bb797dc9f3185a1cae86083e0ff8/media/js/base/fxa-utm-referral.js#L55-L56

Reply:
Yeah, I think letting `getAttributionData` do its thing should be enough for utms 👍